keps/sig-node/1592-pod-level-resource-limits/README.md

```diff
@@ -0,0 +1,439 @@
+then the scheduler will continue to operate as it does in the current implementation. The QoS definitions will be updated to consider the pod-level resources
+when classifying a pod into one of the QoS levels. If the pod level doesn't include any resource limits, then kubelet will continue to function as it does today.
+
+The current implementation will set the cgroup limits for the memory/CPU resources on the pod level of the hierarchy only in the following cases:
+1. The QoS level is `Guaranteed`. 
+1. The QoS level is `Burstable` and all of the containers specify a limit for the relevant resource. 
+
+In the current implementation pods are considered `Guaranteed` if and only if all of the included containers define a resource request which is equal
+to the resource limit. The current implementation considers a pod to be `Burstable` if at least one container defines resource requests or limits. Any
+pods not matching the `Guaranteed` or `Burstable` QoS definitions are placed in the `BestEffort` QoS level.
+
+This proposal suggests modifying these definitions the following: 
+
+A pod is considered `Guaranteed` if either of the following two conditions apply: 
+1. All of the containers included in the pod define memory and CPU resource requests equal to resource limits
+1. The pod includes a pod-level memory and CPU resource request which is equal to the resource limit, even if one or more of the included containers doesn't define either a CPU or a memory request or limit.
+
+A pod is considered `Burstable` if at least one container has a resource request, or the pod itself has a resource request. As before this proposal, setting
+a limit without a request causes Kubernetes to use the limit as a request.
+
+A pod is considered `BestEffort` if there are not resource request or limits settings on any container or on the pod level itself.
+
+If a developer included a request and limit definition at the pod level, the developer's intent was for Kubernetes to attempt to guarantee resources 
+as a whole, for all the containers included in the pod. This should apply even if the developer is unable to preemptively split the resources correctly
+between all the containers contained in the pod. 
+
+Not all resource limits are treated in the same way by the underlying Linux cgroup controllers.  For example, if a memory pod level limit request is smaller 
+than the sum of limit for all the containers, then the limit requested by the containers is unreachable. It becomes impossible for all of the containers to 
+reach their requested limit, since the sum of available memory becomes the pod-level limit. Note that even though setting `pod.limit[memory] < sum(pod.containers[*].limit[memory])`
+would not allow all of the containers to allocate memory up to their limit, this is not considered an error by this proposal (though a warning should be emitted).
```

Comment:
I disagree that a warning is needed.   In this case, the semantic is clear - each container MAY use more memory than requested, but the sum of containers MAY NOT exceed the pod's limit.

Reply:
It might be worth a warning if any single container's limit is > pod limit.   I can't imagine when that would be useful.

Reply:
@thockin Agree to your first comment, I would make the changes in the KEP